src/data/finset/pointwise.lean

```diff
@@ -408,19 +386,76 @@ coe_injective.semigroup _ coe_mul
 protected def comm_semigroup [comm_semigroup α] : comm_semigroup (finset α) :=
 coe_injective.comm_semigroup _ coe_mul
 
+section monoid
+variables [monoid α] {s t : finset α} {a : α}
+
+@[simp, to_additive]
+lemma coe_pow (s : finset α) (n : ℕ) : ↑(s ^ n) = (s ^ n : set α) :=
+begin
+  change ↑(npow_rec n s) = _,
+  induction n with n ih,
+  { rw [npow_rec, pow_zero, coe_one] },
+  { rw [npow_rec, pow_succ, coe_mul, ih] }
+end
+
 /-- `finset α` is a `monoid` under pointwise operations if `α` is. -/
 @[to_additive "`finset α` is an `add_monoid` under pointwise operations if `α` is. "]
-protected def monoid [monoid α] : monoid (finset α) :=
-coe_injective.monoid _ coe_one coe_mul coe_pow
+protected def monoid : monoid (finset α) := coe_injective.monoid _ coe_one coe_mul coe_pow
+
+localized "attribute [instance] finset.mul_one_class finset.add_zero_class finset.semigroup
+  finset.add_semigroup finset.monoid finset.add_monoid" in pointwise
+
+/-- If `u` is a unit in `α`, then `{u}` is a unit in `finset α`. -/
+@[to_additive "If `u` is an additive unit in `α`, then `{u}` is an additive unit in `finset α`.",
+simps coe] protected def _root_.units.finset (u : αˣ) : (finset α)ˣ :=
+{ val := {u},
+  inv := {(u⁻¹ : αˣ)},
+  val_inv := by rw [singleton_mul_singleton, units.mul_inv, singleton_one],
+  inv_val := by rw [singleton_mul_singleton, units.inv_mul, singleton_one] }
```

Comment:
Isn't this `monoid_hom.map_units` or something more generic? You just need `singleton : α ->* finset α`

Reply:
Oh right. Does it still deserve its standalone definition?

Reply:
No, I don't think so; only the `α ->* finset α` map does.